cpu/atmega_common/include/periph_cpu_common.h

```diff
@@ -35,19 +35,19 @@ extern "C" {
  * @{
  */
 #define HAVE_GPIO_T
-typedef uint8_t gpio_t;
+typedef uint16_t gpio_t;
```

Comment:
This is going to have a performance penalty, as the ATmegas are 8bit. I think the defines in `gpio_ext_h` already are using `sizeof(gpio_t)`, so would it be possible to keep that `uint8_t` for 8 bit platforms? (If the cost is that only one GPIO extender could be used there, I bet that users of the ATmegas would take that trade-off. And if not, they could use `CFLAGS += -include <header_with_custom_gpio_t_definition.h> -DHAVE_GPIO_T` to use more GPIO extenders.)

But if this increases maintenance effort and/or lines of code more than a bit, I'd say using 16bit on ATmesags is fine. (The Arduino community are also using `int` (=16 bit) for `digitalRead()` and `digitalWrite()` - likely due to not paying attention when defining the API. And their user base seems to be not upset with that.)

Reply:
I know. But ATmega 2560 defines 10 ports and `GPIO_CPU_PIN(x, y)` is defined by `((x << 4) | y)`. That is, with `uint8_t` there is no space left for the extension flag and extension device id.

Reply:
Maybe then 8 bit unless ATmega2560 is used?

Reply:
Hm, even if have only ATmega 1281, there are 7 ports that require 3 bits for port. With the `#define GPIO_CPU_PIN(x, y)   ((x << 4) | y)` there is space for the extension flag but no space for extension device number. Even for ATmega 328p with 3 ports, there would be only 1 bit left for the extension device number.

The question is whether changing `GPIO_CPU_PIN(x, y)` to `((x << 3) | y)` would be possible. If I'm right, ATmega MCUs have only 8 GPIOs per port.